autodocodec-api-usage/test_resources/openapi-schema/declareSchemaRef/expression.json

```diff
@@ -0,0 +1,92 @@
+{
+    "definitions": {
+        "Expression": {
+            "discriminator": {
+                "mapping": {
+                    "literal": "LiteralExpression",
+                    "product": "ProductExpression",
+                    "sum": "SumExpression"
+                },
+                "propertyName": "type"
+            },
```

Comment:
This is such an ugly feature, but I've found myself needing it so pretty cool that we can support it!